### PR TITLE
query builder fixes

### PR DESCRIPTION
**Pull request by xannem**
## How did you test this change?

- [x] Text editor to light mode
image

- [x]  Misc light mode edits
image
image

- [x] String inputs for filter value now has a button that will allow user more space to input text. If the user's input has multiple lines, the elements will be separated by new lines. If the user input has one line, the inputs will be separated by commas.
Kapture 2022-11-24 at 00 29 58

# Testing

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx [hunk 1]
```diff
@@ -443,9 +443,14 @@ const setCollectionValue = (
     return;
   }
   let result: unknown[] = [];
-  const parseResult = CSVParser.parse<string[]>(value.trim(), {
+  let parseResult = CSVParser.parse<string[]>(value.trim(), {
     delimiter: ',',
   });
+  if (parseResult.data[0]?.length === 1) {
```
Comment (akphi) [inline packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx:449]: we should abstract this out to an utility function and test with some inputs, e.g. `1,3,4,6\n7,8,8,9\n,10`

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx [hunk 2]
```diff
@@ -546,6 +541,8 @@ const CollectionValueInstanceValueEditor = observer(
     const inputRef = useRef<HTMLInputElement>(null);
     const [text, setText] = useState(stringifyValue(valueSpecification.values));
     const [editable, setEditable] = useState(false);
+    const [openedPopUp, setOpenedPopUp] = useState(false);
```
Comment (akphi) [inline packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx:544]: let's rename this:

```suggestion
    const [showAdvancedEditorPopover, setShowAdvancedEditorPopover] = useState(false);
```

File: packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx [hunk 3]
```diff
@@ -579,29 +584,74 @@ const CollectionValueInstanceValueEditor = observer(
...
-          <button
-            className="value-spec-editor__reset-btn"
-            title="Reset"
-            onClick={resetValue}
-          >
-            <RefreshIcon />
-          </button>
-        </div>
+        <>
+          {openedPopUp && (
```
Comment (akphi) [inline packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx:588]: I think we should use a popover here instead. Modal is a little too strong.

File: packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx [hunk 4]
```diff
@@ -579,29 +584,74 @@ const CollectionValueInstanceValueEditor = observer(
...
+              spellCheck={false}
+              value={text}
+              placeholder={text === '' ? '(empty)' : undefined}
+              onChange={changeValue}
+            />
+            <button
+              className="lambda-editor__action"
+              onClick={() => setOpenedPopUp(true)}
+              tabIndex={-1}
+              title="Open..."
```
Comment (akphi) [inline packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx:636]: Please change the title tooltip text to something more descriptive

File: packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx [hunk 5]
```diff
@@ -579,29 +584,74 @@ const CollectionValueInstanceValueEditor = observer(
...
+              className={clsx(
+                'panel__content__form__section__input value-spec-editor__input',
+              )}
+              spellCheck={false}
+              value={text}
+              placeholder={text === '' ? '(empty)' : undefined}
+              onChange={changeValue}
+            />
+            <button
+              className="lambda-editor__action"
```
Comment (akphi) [inline packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx:633]: why are we using this class here?

File: packages/legend-art/style/base/_modal.scss [hunk 6]
```diff
@@ -170,3 +170,37 @@
     height: calc(100% - 3.6rem);
   }
 }
+
+.editor-modal--small {
```
Comment (akphi) [inline packages/legend-art/style/base/_modal.scss:174]: If we are to use the popover as suggested below, we don't need this change.

**Review by akphi — CHANGES_REQUESTED** [review]

**Comment by akphi** [comment]
Also, @xannem let's include the UI change you made to the OLAP panel (light/dark mode). Also, can you fix a minor issue where in query builder, when we open the result set modifier (under TDS panel), the take limit input has no visible outline in light theme?

File: packages/legend-shared/src/CommonUtils.ts [hunk 7]
```diff
@@ -337,6 +338,47 @@ export const deleteEntry = <T>(
   return false;
 };
 
+/**
+ * NOTE: this splits a string value into an array of strings by using a
```
[the remark this answers is not in the document]
Comment (xannem) [reply packages/legend-shared/src/CommonUtils.ts:342]: Yep that's what I believe we both discussed should be the case for now 🙇

File: packages/legend-shared/package.json [hunk 8]
```diff
@@ -43,6 +43,7 @@
     "@types/lodash-es": "4.17.6",
     "@types/object-hash": "2.2.1",
     "@types/pako": "2.0.0",
+    "@types/papaparse": "5.3.5",
```
[the remark this answers is not in the document]
Comment (xannem) [reply packages/legend-shared/package.json:46]: It is used in legend-shared but still needed in legend-query-builder for `stringifyValue` for `ValueSpecification` things 🙇

[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-shared/package.json:46]: then let's move that method over to `@finos/legend-shared` as well

File: packages/legend-shared/src/CommonUtils.ts [hunk 9]
```diff
@@ -337,6 +338,47 @@ export const deleteEntry = <T>(
...
+ * delimiter of a comma if the string is only one line. However, if the
+ * string has multiple lines, the delimiter will not be applied. This is so
+ * that for an example input of multiple lines like
+ *       One, Comma in One
+ *       Two
+ *       Three
+ * will still equal 3 elements (['One, Comma in One', 'Two', 'Three']) rather than 4
+ */
+export const parseCSVString = (value: string): string[] | undefined => {
```
Comment (akphi) [inline packages/legend-shared/src/CommonUtils.ts:351]: This should be in `FormatterUtils.ts`, not here.

File: packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx [hunk 10]
```diff
@@ -601,29 +603,61 @@ const CollectionValueInstanceValueEditor = observer(
...
-          <button
-            className="value-spec-editor__reset-btn"
-            title="Reset"
-            onClick={resetValue}
-          >
-            <RefreshIcon />
-          </button>
-        </div>
+        <>
+          {showAdvancedEditorPopover && (
```
Comment (akphi) [inline packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx:607]: Looking at the video preview, I think when we're showing the popover, we should not show anything in the `input` field, it looks odd, you can even consider moving the popover to completely eclipse the original input box, by manipulating `menuProps.anchorOrigin and menuProps.transformOrigin`. Also, add a small box at the bottom of the popover to say - hit `Enter` to apply change, `Esc` and `click-out` action should not save the result <-- this is also the behavior in `vscode`, look at vscode when you try to create a new file in the explorer tree sidebar, that's kind of the visual guide and interaction I hope we have.

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-shared/package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (xannem) [reply packages/legend-shared/package.json:46]: The `stringifyValue` method uses CVSParser to unparse but checks if the value is instance of `PrimitiveInstanceValue` or `EnumValueInstanceValue` and takes in `ValueSpecification[]`, so I did originally try moving it but then seeing as those are all types that didn't belong in `legend-shared` and shouldn't be imported I thought maybe the method was ok to be left here?

**Comment by MauricioUyaguari** [comment]
@xannem lets bring this PR back to life. So many great changes here :)

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**